bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Cursor.java

```diff
@@ -319,6 +338,55 @@ public Cursor(Device device, ImageData source, int hotspotX, int hotspotY) {
 	init();
 }
 
+/**
+ * <b>IMPORTANT:</b> This method is not part of the public
+ * API for Image. It is marked public only so that it
+ * can be shared within the packages provided by SWT. It is not
+ * available on all platforms, and should never be called from
+ * application code.
+ *
+ * Updates zoom and refresh the Cursor based on the native zoom level, if required.
+ *
+ * @param cursor the cursor to get the handle of
+ * @param zoom device zoom in % of the monitor on which the image is painted
+ *
+ * @return true if cursor is refreshed
+ *
+ * @noreference This method is not intended to be referenced by clients.
+ */
+public static Long win32_getHandle (Cursor cursor, int zoom) {
+	if (cursor.isDisposed()) {
+		return cursor.handle;
+	}
+	if (cursor.zoomLevelToHandle.get(zoom) != null) {
+		return cursor.zoomLevelToHandle.get(zoom);
+	}
+
+	if (cursor.source == null) {
+		cursor.setHandleForZoomLevel(cursor.handle, zoom);
+	} else {
+		ImageData source = DPIUtil.scaleImageData(cursor.device, cursor.source, zoom, DEFAULT_ZOOM);
+		if (cursor.isIcon) {
+			Cursor newCursor = new Cursor(cursor.device, source, cursor.hotspotX, cursor.hotspotY);
+			cursor.setHandleForZoomLevel(newCursor.handle, zoom);
+		} else {
+			ImageData mask = DPIUtil.scaleImageData(cursor.device, cursor.mask, zoom, DEFAULT_ZOOM);
+			Cursor newCursor = new Cursor(cursor.device, source, mask, cursor.hotspotX, cursor.hotspotY);
+			cursor.setHandleForZoomLevel(newCursor.handle, zoom);
+		}
```

Comment:
```suggestion
		ImageData mask = cursor.isIcon ? DPIUtil.scaleImageData(cursor.device, cursor.mask, zoom, DEFAULT_ZOOM) : null;
		Cursor newCursor = new Cursor(cursor.device, source, mask, cursor.hotspotX, cursor.hotspotY);
		cursor.setHandleForZoomLevel(newCursor.handle, zoom);
```

Reply:
This suggestion won't work as we are using different constructor for both the conditions. It's not just about using the mask imageData

Reply:
Hm, you're right. I assumed that the constructors would be fairly similar but they are not.

Scratch my suggestion.